ts/store/reducers/backendStatus.ts

```diff
@@ -191,6 +191,12 @@ export const isFIMSEnabledSelector = createSelector(
     backendStatus.map(bs => bs.config.fims.enabled).toUndefined() ?? false
 );
 
+export const fimsDomainSelector = createSelector(
+  backendStatusSelector,
+  (backendStatus): string | undefined =>
+    backendStatus.map(bs => bs.config.fims.domain).toUndefined()
```

Comment:
I noticed that if `domain` is added without the protocol (`http://` or `https://`, so just `127.0.0.1` for example) the app is going to crash:

https://user-images.githubusercontent.com/5963683/168297997-e64b3718-feba-419d-8762-f91725fa406c.mov

What do you think about adding here a check, returning `undefined` if the given domain is without a protocol? Something like a simple regex (I.e. `^https?:\/\/`) could be fine, or I don't know if inside the app we already have something to validate a URL 🤔

Do you think it's worth it?

Reply:
That should work too! 😃

Reply:
[911a34b](https://github.com/pagopa/io-app/pull/3871/commits/911a34b6926a8033031091f8079754a178203756) deal! ✅

Reply:
@CrisTofani Sadly `canOpenURL` returns a Promise, I missed this thing I thought it was sync 😕 I still get the crash whenever I use `127.0.0.1` as a domain instead of `http://127.0.0.1`. I just noticed that we are parsing that url in the "next" step `const url = new URLParse(maybeFimsDomain.value as string, true);`, what about checking for the `protocol` field there?

A possible change could be:

```ts
// ...

const maybeParsedUrl = maybeFimsDomain.chain(domain => {
  const parsed = new URLParse(domain as string, true);
  return parsed.protocol ? some(parsed) : none;
});

if (maybeParsedUrl.isNone()) {
  Alert.alert(I18n.t("global.genericAlert"), "", [
    {
      text: I18n.t("global.buttons.exit"),
      style: "default",
      onPress: goBackAndResetInternalNavigationInfo
    }
  ]);
  return;
}

// ...
```

What do you think? 🙂